odoo/fields.py

```diff
@@ -1536,11 +1537,14 @@ class _String(Field):
     translate = False                   # whether the field is translated
     prefetch = None
     unaccent = True
+    pattern = (None, None, None)        # tuple (description, regex) to validate an input
 
     def __init__(self, string=Default, **kwargs):
         # translate is either True, False, or a callable
         if 'translate' in kwargs and not callable(kwargs['translate']):
             kwargs['translate'] = bool(kwargs['translate'])
+        if 'pattern' in kwargs and len(kwargs['pattern']) == 2:
+            kwargs['pattern'] = (*kwargs['pattern'], 0)
```

Comment:
I wonder whether flags should be explit or not.  In my experience, that fact that flags are optional in module `re` often make them incorrect in practice.  I found regexpes in my own code that lacked either the case-insensitive or multi-line flag, simply because I didn't thought about the flags when writing it.  But that's only my opinion.

Reply:
People would just write 0 instead of verifying that they indeed don't need any flag. It is optional in most libs, we should be consistent.

Reply:
> People would just write 0 instead of verifying that they indeed don't need any flag.

That is what I ended up doing (but it is consistent since no flags were used before for the fields I translated with the new param)